docs/rules/prefer-string-starts-ends-with.md

```diff
@@ -4,6 +4,8 @@ Prefer [`String#startsWith()`](https://developer.mozilla.org/en/docs/Web/JavaScr
 
 This rule is fixable.
 
+Note: the autofix will throw an exception when the string being tested is `null` or `undefined`. Several safer but more verbose automatic suggestions are provided for this situation.
```

Comment:
It not only about `null` and `undefined`, could be anything that's not a string

```js
/^1/.test(1);
```

```js
/^t/.test(true);
```

```js
/^1/.test(10n);
```

```js
/^1,2/.test([1,2,3,])
```

```js
/^\[object/.test({})
```

```js
/^a/.test(async () => {})
```

Reply:
Updated messaging.